handlers/handler_api.go

```diff
@@ -0,0 +1,93 @@
+package handlers
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+// RegisterSQSHandler registers endpoints
+func (handler *SQSHandler) RegisterSQSHandler() {
+	http.HandleFunc("/server", handler.ServiceHandler)
+	http.HandleFunc("/sqs", handler.SQSHandler)
+	http.HandleFunc("/job", handler.HandleJob)
+}
+
+// ServiceHandler handles stop/start the SQS querrying service
+func (handler *SQSHandler) ServiceHandler(w http.ResponseWriter, r *http.Request) {
+	if r.Method == "PUT" {
+		val := r.URL.Query().Get("start")
+		if val == "true" {
+			handler.StartServer()
+		} else if val == "false" {
+			handler.ShutdownServer()
+		}
+
+	} else {
+		http.Error(w, "Not supported request method.", 405)
+	}
+
+}
+
+// SQSHandler handles update/get SQS URL string
+func (handler *SQSHandler) SQSHandler(w http.ResponseWriter, r *http.Request) {
+	if r.Method == "PUT" {
+		handler.QueueURL = r.URL.Query().Get("url")
+		handler.ShutdownServer()
+		handler.StartServer()
+
+	} else if r.Method == "GET" {
+		fmt.Fprintf(w, handler.QueueURL)
+	} else {
+		http.Error(w, "Not supported request method.", 405)
+	}
+}
+
+func (handler *SQSHandler) HandleJob(w http.ResponseWriter, r *http.Request) {
+	if r.Method == "GET" {
+		handler.listJob(w, r)
+	} else if r.Method == "POST" {
+		handler.addJob(w, r)
+	} else if r.Method == "DELETE" {
+		handler.deleteJob(w, r)
+	}
+}
+
+// addImagePatternMap add an pattern-image map
+func (handler *SQSHandler) addJob(w http.ResponseWriter, r *http.Request) {
+	// An example of an json body
+	//
+	// {
+	// 	"name": "usersync",
+	// 	"pattern": "s3://user_bucket/",
+	// 	"image": "quay.io/cdis/fence:master",
+	// 	"imageConfig": {
+	//	   "indexURL": "http://index-service/"
+	//	}
+	// }
+	//
+	// Try to read the request body.
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		msg := fmt.Sprintf("failed to read request body; encountered error: %s", err)
+		http.Error(w, msg, http.StatusBadRequest)
+		return
+	}
+	handler.addNewJobConfig(body)
```

Comment:
I'm still not sure what's the best way to approach this...but you might consider doing the parsing here, for example define a `JobConfig` struct and right here you read the bytes into that, and `handler.addNewJobConfig` accepts a `JobConfig` struct instead of bytes. Just some thoughts, to me that seems to end up with a cleaner interface

Reply:
I don't want the function body is too long and carries too many jobs. This function simply extract the json body and give to other function to handle. Byte is most basic primitive type in Golang just like int, string etc.. So I feel comfortable with passing it